fuse/data/datasets/caching/samples_cacher.py

```diff
@@ -207,14 +207,16 @@ def cache_samples(self, orig_sample_ids: List[Any]) -> List[Tuple[str, Union[Non
             orig_sid_to_final[initial_sample_id] = output_sample_ids
 
         write_dir = self._get_write_dir()
+
+        set_info_dir = os.path.join(write_dir, "full_sets_info")
+        os.makedirs(set_info_dir, exist_ok=True)
+
         pipeline_desc_file = os.path.join(write_dir, f"pipeline_{self._pipeline_desc_hash}_desc.txt")
```

Comment:
pipeline_desc_file is not in set_info_dir.
What was your error?

Reply:
makedirs create all the directories in the path to the root-dir.